### ISSUE TITLE
HPO: gpu_metrics file does not exist!

### ISSUE THREAD

**Issue by MinYounZhang** [microsoft/nni#4788]
**Describe the issue**:
I run`nnictl create --config exp.yaml --port 9090` and  cheak `nvidia-smi`, find its running without using GPU.
So  I run`nnictl create --config exp.yaml --port 9090 - f` and get error below:
![image](https://user-images.githubusercontent.com/19604387/164359111-4e2dd2ac-33ae-466f-b21d-774ff51977af.png)

I have followed tips in https://github.com/microsoft/nni/issues/4226
`METRIC_OUTPUT_DIR=${scriptFolder} python3 -m nni.tools.gpu_tool.gpu_metrics_collector` works under my path.

**Environment**:
- NNI version: 2.6
- Training service (local|remote|pai|aml|etc): local
- Client OS: Ubuntu 18.04
- Server OS (for remote mode only): \
- Python version:  py3.7
- PyTorch/TensorFlow version:  pytorch 1.10.2
- Is conda/virtualenv/venv used?: Y
- Is running in Docker?: N

**Configuration**:
 - Experiment config (remember to remove secrets!):
 - Search space:
![image](https://user-images.githubusercontent.com/19604387/164230399-10b95038-ce0d-4e3b-b6d5-1f118a78e76f.png)

**Log message**:
 - nnimanager.log:
 Dont exist
 - dispatcher.log:
[2022-04-20 20:12:10] INFO (nni.experiment/MainThread) Creating experiment, Experiment ID: 30pecn7s
[2022-04-20 20:12:10] INFO (nni.experiment/MainThread) Starting web server...
[2022-04-20 20:12:11] WARNING (nni.experiment/MainThread) Timeout, retry...
[2022-04-20 20:12:12] INFO (nni.experiment/MainThread) Setting up...
[2022-04-20 20:12:12] INFO (nni.experiment/MainThread) Web UI URLs: http://127.0.0.1:9090 http://172.17.7.99:9090 http://172.18.0.1:9090 http://172.19.0.1:9090 http://172.20.0.1:9090
[2022-04-20 20:12:12] INFO (nni.tools.nnictl.launcher/MainThread) To stop experiment run "nnictl stop 30pecn7s" or "nnictl stop --all"
[2022-04-20 20:12:12] INFO (nni.tools.nnictl.launcher/MainThread) Reference: https://nni.readthedocs.io/en/stable/Tutorial/Nnictl.html
[2022-04-20 20:12:12] INFO (nni.experiment/MainThread) Stopping experiment, please wait...
[2022-04-20 20:12:12] INFO (nni.tuner.tpe/MainThread) Using random seed 1504156515
[2022-04-20 20:12:12] INFO (nni.runtime.msg_dispatcher_base/MainThread) Dispatcher started
[2022-04-20 20:12:12] INFO (nni.runtime.msg_dispatcher_base/MainThread) Dispatcher exiting...
[2022-04-20 20:12:15] INFO (nni.runtime.msg_dispatcher_base/MainThread) Dispatcher terminiated
 - nnictl stdout and stderr:
 Dont exist

**How to reproduce it?**:
nnictl create --config exp.yaml --port 9090 - f

**Comment by liuzhe-lz** [comment]
Does the Python script create `/tmp/USER/nni/script/gpu_metrics` file?

**Comment by MinYounZhang** [comment]
> Does the Python script create `/tmp/USER/nni/script/gpu_metrics` file?

Nothing under  `/tmp/USER/nni`.

**Comment by MinYounZhang** [comment]
> Does the Python script create `/tmp/USER/nni/script/gpu_metrics` file?

Nothing under `tmp/USER/nni`.
I create `/tmp/USER/nni/script/gpu_metrics` myself.
Run `nnictl create --config exp.yaml -p 9090 -f` and get the following error log:
![image](https://user-images.githubusercontent.com/19604387/164384216-5b97ceb7-de9c-4d0d-b1c6-91b96eff0e06.png)

**Comment by liuzhe-lz** [comment]
By manually running `METRIC_OUTPUT_DIR=/tmp/zmy/nni/script python3 -m nni.tools.gpu_tool.gpu_metrics_collector`, what's the output in console and in the directory?
It's a relatively simple script: https://github.com/microsoft/nni/blob/master/nni/tools/gpu_tool/gpu_metrics_collector.py
If does not output anything at all, could you please help debug it?

**Comment by MinYounZhang** [comment]
> By manually running `METRIC_OUTPUT_DIR=/tmp/zmy/nni/script python3 -m nni.tools.gpu_tool.gpu_metrics_collector`, what's the output in console and in the directory? It's a relatively simple script: https://github.com/microsoft/nni/blob/master/nni/tools/gpu_tool/gpu_metrics_collector.py If it does not output anything at all, could you please help debug it?

`cat /tmp/USER/nni/script/gpu_metrics`

{"Timestamp": "Thu Apr 21 15:25:23 2022", "gpuCount": 4, "gpuInfos": [{"activeProcessNum": 0, "gpuMemUtil": "0", "gpuUtil": "0", "index": 0}, {"activeProcessNum": 0, "gpuMemUtil": "0", "gpuUtil": "0", "index": 1}, {"activeProcessNum": 0, "gpuMemUtil": "0", "gpuUtil": "0", "index": 2}, {"activeProcessNum": 0, "gpuMemUtil": "0", "gpuUtil": "0", "index": 3}]}

**Comment by liuzhe-lz** [comment]
After some investigation I think the warning is not the problem. It periodically queries GPU metrics and at first query the file is not ready yet.
The real problem is tuner exited immediately after launching.
```
[2022-04-20 20:12:12] INFO (nni.experiment/MainThread) Stopping experiment, please wait...
```

Could you use `nnictl create ... --debug` to launch the experiment and upload `dispatcher.log` & `nnimanager.log`?

**Comment by MinYounZhang** [comment]
> After some investigation I think the warning is not the problem. It periodically queries GPU metrics and at first query the file is not ready yet. The real problem is tuner exited immediately after launching.
> 
> ```
> [2022-04-20 20:12:12] INFO (nni.experiment/MainThread) Stopping experiment, please wait...
> ```
> 
> Could you use `nnictl create ... --debug` to launch the experiment and upload `dispatcher.log` & `nnimanager.log`?

thanks for ur  reply :)
[dispatcher.log](https://github.com/microsoft/nni/files/8560781/dispatcher.log)
[nnimanager.log](https://github.com/microsoft/nni/files/8560787/nnimanager.log)
